segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/AppendProcessor.java

```diff
@@ -161,14 +163,32 @@ public void setupAppend(SetupAppend setupAppend) {
                 .whenComplete((attributes, u) -> {
                     try {
                         if (u != null) {
+                            if (u instanceof StreamSegmentSealedException) {
+                                throw u;
+                            }
                             handleException(writer, setupAppend.getRequestId(), newSegment, "setting up append", u);
                         } else {
+                            SegmentProperties sp = this.store.getStreamSegmentInfo(newSegment, TIMEOUT).join();
```

Comment:
Invoking `join` here will eventually lead to a thread pool exhaustion.

Reply:
Also why are you doing this for successful operations?

Reply:
Yes Andrei, I could not find SegmentSealed exception raised until here.